### PR TITLE
chore(deps): update Cargo.lock

### PR DESCRIPTION
**Pull request by hanabi1224**
## Description

- run `cargo update`
- lock `webrtc-ice = "=0.10.0"` to not break webrtc smoke tests
- fix `cargo clippy` warnings
- update `deny.toml` accordingly

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by drHuangMHT** [comment]
Dependencies are managed by bots automatically, I don't think it is a good idea to just run `cargo update` and then done. cc @jxs

**Comment by hanabi1224** [comment]
@drHuangMHT As I understand, dependabot manages only direct dependencies, while `cargo update` bumps all indirect dependencies as well. Also, since `libp2p` works as libraries, having `Cargo.lock` in source control could [give some false sense of security](https://doc.rust-lang.org/cargo/faq.html#why-have-cargolock-in-version-control), it's important to regularly update all indirect dependencies.  e.g. `libp2p-webrtc` smote test scenario does not work without locking `webrtc-ice = "=0.10.0"`, or exclude `Cargo.lock` from source control.

> However, this determinism can give a false sense of security because Cargo.lock does not affect the consumers of your package, only Cargo.toml does that.

File: deny.toml [hunk 1]
```diff
@@ -44,6 +44,7 @@ allow = [
     "MPL-2.0",
     "Unlicense",
     "Unicode-3.0",
+    "Zlib",
```
Comment (jxs) [inline deny.toml]: do we need to adopt this license? If so I'd suggest we do it in a different PR as we did with the Unicode one on  https://github.com/libp2p/rust-libp2p/pull/5738

**Review by jxs — COMMENTED** [review]
Thanks for this @hanabi1224!
 
> @drHuangMHT As I understand, dependabot manages only direct dependencies, while `cargo update` bumps all indirect dependencies as well. Also, since `libp2p` works as libraries, having `Cargo.lock` in source control could [give some false sense of security](https://doc.rust-lang.org/cargo/faq.html#why-have-cargolock-in-version-control), it's important to regularly update all indirect dependencies. e.g. `libp2p-webrtc` smote test scenario does not work without locking `webrtc-ice = "=0.10.0"`, or exclude `Cargo.lock` from source control.
> 
> > However, this determinism can give a false sense of security because Cargo.lock does not affect the consumers of your package, only Cargo.toml does that.

yeah this PR is helpful due to also the freeze in the `webrc-ice` freeze, can you just address the comment?
Thanks!

File: deny.toml (hunk 1 above)
Comment (hanabi1224) [reply deny.toml]: Reverted. Please note that CI fails without this change: https://github.com/libp2p/rust-libp2p/actions/runs/12519077957/job/34922566527?pr=5755#step:4:25868

Comment (jxs) [reply deny.toml]: no worries, we can address it on a subsequent PR :)

**Review by jxs — APPROVED** [review]
